packages/app-mobile/components/screens/Note/Note.tsx

```diff
@@ -1439,6 +1389,12 @@ class NoteScreenComponent extends BaseScreenComponent<ComponentProps, State> imp
 	}
 
 	public render() {
+		// Commands must be registered before child components can render.
+		// Calling this in the constructor won't work in strict mode, where
+		// componentWillUnmount (which removes the commands) can be called
+		// multiple times.
+		this.registerCommands();
```

Comment:
I don't like calling `this.registerCommands` in `render` &mdash; intuitively, the constructor or `componentDidMount` seem to make more sense. However, both of these other options have potential issues:
- **componentDidMount**: If commands are registered in `componentDidMount`, creating a new note causes `ToolbarButtonUtils` to throw an `Error` (runtime not registered). In this case, `componentDidMount` runs after the first render of the editor toolbar. The editor toolbar (via ToolbarButtonUtils) requires that all commands have registered runtimes.
- **constructor**: Calling `registerCommands` from the `constructor` isn't sufficient if `componentWillUnmount` is called before the last render (e.g. if `componentDidMount` is called, then `componentDidUnmount`, then `componentDidMount`, then `render`). If this happens, all renders that happen after the first call to `componentWillUnmount` may fail due to unregistered command runtimes. According to the React documentation, this can happen in strict mode:
    > When [Strict Mode](https://react.dev/reference/react/StrictMode) is on, in development React will call [componentDidMount,](https://react.dev/reference/react/Component#componentdidmount) then immediately call componentWillUnmount, and then call componentDidMount again. This helps you notice if you forgot to implement componentWillUnmount or if its logic doesn’t fully “mirror” what componentDidMount does.
    >
    > — https://react.dev/reference/react/Component#componentwillunmount-caveats

    I'm unsure whether `componentWillUnmount` can be called multiple times when not in strict mode. However, based on the above documentation, `componentWillUnmount` should be the "mirror" of `componentDidMount`.  If command runtimes are registered in the `constructor` and unregistered in `componentWillUnmount`, then this is not the case.

Migrating `Note.tsx` to a function component may allow simplifying this logic &mdash; function components that do similar things currently use [`useNowEffect`](https://github.com/laurent22/joplin/blob/dev/packages/lib/hooks/useNowEffect.ts) to register commands.